pkg/blockstorage/blockstorage.go

```diff
@@ -29,6 +29,7 @@ type Provider interface {
 	VolumeGet(ctx context.Context, id string, zone string) (*Volume, error)
 	// Snapshot operations
 	SnapshotCopy(ctx context.Context, from Snapshot, to Snapshot) (*Snapshot, error)
+	SnapshotCopyWithArgs(ctx context.Context, from Snapshot, to Snapshot, args map[string]string) (*Snapshot, error)
```

Comment:
Could you please add a comment explaining why this is different from SnapshotCopy and why one should use one over the other...